tests/io/test_api.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+from pytest import mark
+import aeon.io.api as aeon
+from aeon.schema.dataset import exp02
+import pandas as pd
+
+@mark.api
+def test_load_start_only():
+    data = aeon.load(
+        './tests/data/nonmonotonic',
+        exp02.Patch2.Encoder,
+        start=pd.Timestamp('2022-06-06T13:00:49'))
+    assert len(data) > 0
+
+@mark.api
+def test_load_end_only():
+    data = aeon.load(
+        './tests/data/nonmonotonic',
+        exp02.Patch2.Encoder,
+        end=pd.Timestamp('2022-06-06T13:00:49'))
+    assert len(data) > 0
+
+@mark.api
+def test_load_filter_nonchunked():
+    data = aeon.load(
+        './tests/data/nonmonotonic',
+        exp02.Metadata,
+        start=pd.Timestamp('2022-06-06T09:00:00'))
+    assert len(data) > 0
+
+@mark.api
+def test_load_monotonic():
+    data = aeon.load('./tests/data/monotonic', exp02.Patch2.Encoder)
+    assert data.index.is_monotonic_increasing
+
+@mark.api
+def test_load_nonmonotonic():
+    data = aeon.load('./tests/data/nonmonotonic', exp02.Patch2.Encoder)
+    assert not data.index.is_monotonic_increasing
```

Comment:
```suggestion
@pytest.mark.parametrize(
    "args,kwargs,expected",
    [
        pytest.param(
            ["./tests/data/nonmonotonic", exp02.Patch2.Encoder],
            {"start": pd.Timestamp("2022-06-06T13:00:49")},
            0,
            id="start_only",
            marks=mark.api,
        ),
        pytest.param(
            ["./tests/data/nonmonotonic", exp02.Patch2.Encoder],
            {"end": pd.Timestamp("2022-06-06T13:00:49")},
            0,
            id="end_only",
            marks=mark.api,
        ),
        pytest.param(
            ["./tests/data/nonmonotonic", exp02.Metadata],
            {"start": pd.Timestamp("2022-06-06T09:00:00")},
            0,
            id="non_chunked",
            marks=mark.api,
        ),
        pytest.param(
            ["./tests/data/monotonic", exp02.Patch2.Encoder],
            {},
            True,
            id="monotonic",
            marks=mark.api,
        ),
        pytest.param(
            ["./tests/data/nonmonotonic", exp02.Patch2.Encoder],
            {},
            False,
            id="nonmonotonic",
            marks=mark.api,
        ),
    ],
)
def test_load(args, kwargs, expected):
    data = aeon.load(*args, **kwargs)
    if type(expected) == bool:
        assert data.index.is_monotonic_increasing == expected
    else:
        assert len(data) > expected
```

Since these tests are testing the same ``load`` function with various inputs, we could also consider [parameterised tests](https://docs.pytest.org/en/latest/how-to/parametrize.html#parametrize-basics).

Reply:
I feel parameterised tests work best when each set of parameters fits in a line so you can massively reduce duplication. In this case however since there are so many different parameters to vary it looks like the parameterised version takes up even more lines than the original code.